tests/conn_can/main.c

```diff
@@ -613,8 +637,83 @@ static void *_receive_thread(void *args)
     return NULL;
 }
 
+static int help(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    puts("Help:");
+    puts("\tinit [trx_id] - initialize a trx");
+    puts("\tset_mode [trx_id] [mode] - set a mode on the trx");
+    printf("trx_id: 0..%u\n", (unsigned)ARRAY_SIZE(devs));
+    puts("modes:");
+    puts("\t0: normal mode");
+    puts("\t1: silent mode");
+    puts("\t2: standby mode");
+    puts("\t3: high-speed mode (SW CAN only)");
+    puts("\t4: high-voltage wakeup mode (SW CAN only)");
+
+    return 0;
+}
+
+static int init(int argc, char **argv) {
+
+    if (argc < 2) {
+        puts("trx_id needed");
+        help(0, NULL);
+        return 1;
+    }
+
+    unsigned trx = atoi(argv[1]);
+    if (trx < ARRAY_SIZE(devs)) {
+        int res = can_trx_init(devs[trx]);
+        if (res >= 0) {
+            puts("Trx successfully initialized");
+            return 0;
+        }
+        else {
+            printf("Error when initializing trx: %d\n", res);
+        }
+    }
+    else {
+        puts("Invalid trx_id");
+    }
+
+    return 1;
+}
+
+static int set_mode(int argc, char **argv) {
+
+    if (argc < 3) {
+        puts("trx_id and mode needed");
+        help(0, NULL);
+        return 1;
+    }
+    unsigned trx = atoi(argv[1]);
+    unsigned mode = atoi(argv[2]);
+    if ((trx < ARRAY_SIZE(devs)) &&
```

Comment:
Better to just print error messages and return instead of making larger nesting.